### PR TITLE
On-ramp: Upgrade on-ramp-sdk to 1.9.0

### PR DESCRIPTION
**Pull request by wachunei**
**Development & PR Process**
1. Follow MetaMask [Mobile Coding Standards](https://docs.google.com/document/d/1VJLwTRsUw_5EDq_o8d6sSbXUAYENLurkRitYO45eY5o/edit?usp=sharing)
2. Add `release-xx` label to identify the PR slated for a upcoming release (will be used in release discussion)
3. Add `needs-dev-review` label when work is completed
4. Add `needs-qa` label when dev review is completed
5. Add `QA Passed` label when QA has signed off

**Description**

This PR upgrades on-ramp-sdk to version 1.9.0, removes the use of `getOrderIdFromCallback` in favor of `getOrderFromCallback`. 

**Screenshots/Recordings**

~~ _If applicable, add screenshots and/or recordings to visualize the before and after of your change_~~

**Issue**

Progresses #???

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

### CODE REVIEW

File: app/components/UI/FiatOnRampAggregator/Views/Checkout.tsx [hunk 1]
```diff
@@ -185,12 +182,12 @@ const CheckoutWebView = () => {
   const handleNavigationStateChange = async (navState: WebViewNavigation) => {
     if (
       !isRedirectionHandled &&
-      navState?.url.startsWith(callbackBaseUrl) &&
+      navState.url.startsWith(callbackBaseUrl) &&
```
Comment (NicolasMassart) [inline app/components/UI/FiatOnRampAggregator/Views/Checkout.tsx:185]: Q: Is this part of the required changes for the upgrade or bug fixes?

File: app/components/UI/FiatOnRampAggregator/Views/GetQuotes.tsx [hunk 2]
```diff
@@ -372,8 +372,8 @@ const GetQuotes = () => {
               amountOut: acc.amountOut + (curr?.amountOut || 0),
               totalFee,
               totalGasFee: acc.totalGasFee + (curr?.networkFee || 0),
-              totalProccessingFee:
-                acc.totalProccessingFee + (curr?.providerFee || 0),
+              totalProcessingFee:
```
Comment (NicolasMassart) [inline app/components/UI/FiatOnRampAggregator/Views/GetQuotes.tsx:375]: Comment: thanks for making my spell checker happy!

File: app/components/UI/FiatOnRampAggregator/hooks/useInAppBrowser.ts [hunk 3]
```diff
@@ -56,7 +55,7 @@ export default function useInAppBrowser() {
       // @ts-expect-error navigation prop mismatch
       navigation.dangerouslyGetParent()?.pop();
       NotificationManager.showSimpleNotification(
-        getNotificationDetails(transformedOrder as any),
+        getNotificationDetails(transformedOrder),
```
Comment (NicolasMassart) [inline app/components/UI/FiatOnRampAggregator/hooks/useInAppBrowser.ts:58]: Comment: I like this, cleaner.

File: app/components/UI/FiatOnRampAggregator/hooks/useInAppBrowser.ts [hunk 4]
```diff
@@ -115,16 +114,8 @@ export default function useInAppBrowser() {
...
-
-          if (result.type === 'success' && result.url) {
-            orders = await SDK.orders();
-            orderId = await orders.getOrderIdFromCallback(
-              provider.id,
-              result.url,
-            );
-          } else {
+
+          if (result.type !== 'success' || !result.url) {
```
Comment (NicolasMassart) [inline app/components/UI/FiatOnRampAggregator/hooks/useInAppBrowser.ts:118]: C: 👍

**Review by NicolasMassart — COMMENTED** [review]
Nice work. Cleaner and easier to understand code in the end.

File: app/components/UI/FiatOnRampAggregator/Views/Checkout.tsx (hunk 1 above)
Comment (wachunei) [reply app/components/UI/FiatOnRampAggregator/Views/Checkout.tsx:185]: Just removing an unnecessary check

**Review by tommasini — APPROVED** [review]
LGTM

@wachunei Just a question I notice that you added the label QA passed, is that right?

**Pull request merged**